Feb. 2021 News (#16898)
idaholab/moose: modules/doc/content/newsletter/2021_02.md

container type is hardly ever slower (and if it is, it is by a very small
amount) and in some cases is much faster than the previous default AD container
time (locally indexed non-sparse). This change in default configuration should
hopefully be little-felt by users. However, for "power" AD developers, this
change may open doors to more complex calculations. For example in finite volume
gradient reconstruction, neighbor of neighbor information is required. Handling
this larger element/cell stencil with a locally indexed container would be
extremely difficult; however, it is very straightforward and natural using the
globally indexed container. Here global vs. local means indexing based on
globally numbered degrees of freedom vs. locally numbered degrees of freedom;
one can hopefully imagine that as residuals become more and more dependent on non-local
information, a local indexing scheme becomes more and more untenable.

## GridPartitioner Improvement

[GridPartitioner.md] creates a uniform processor grid that overlays the mesh to be partitioned 
and then assigns all elements of the to-be-partitioned mesh within each cell of the grid

> to the same processor. Previously, a native MOOSE mesh was explicitly created to represent
> the uniform processor grid, and this could cause issues for a specific scenario.  This 
> issue has been addressed by creating a virtual grid instead of a native moose mesh, and all 

Reviewer: ```suggestion
to the same processor. Previously, a native MOOSE mesh was explicitly created to represent
the uniform processor grid, and this could cause issues for a specific scenario where the native 
MOOSE mesh interfered with the attachment of geometric relationship managers.  This 
issue has been addressed by creating a virtual grid instead of a native MOOSE mesh, and all
```

Reviewer: @fdkong

> [GridPartitioner.md] creates a uniform processor grid that overlays the mesh to be partitioned.
> Then it assigns all elements of the to-be-partitioned mesh within each cell of the grid
> to the same processor. We used to explicitly create a native MOOSE mesh to represent
> the uniform processor grid. It caused issues for a specific scenario.  Recently, we
> improved this issue by creating a virtual grid instead of a native moose mesh.
> All operations for mesh element assignments are done using the virtual temporary grid only.

Reviewer: ```suggestion
[GridPartitioner.md] creates a uniform processor grid that overlays the mesh to be partitioned 
and then assigns all elements of the to-be-partitioned mesh within each cell of the grid
to the same processor. Previously, a native MOOSE mesh was explicitly created to represent
the uniform processor grid, and this could cause issues for a specific scenario.  This 
issue has been addressed by creating a virtual grid instead of a native moose mesh, and all 
operations for mesh element assignments are done using the virtual temporary grid only.
```
Here's a suggestion to re-work the wording here. Could you also elaborate (if possible) on that the `specific scenario` was that prompted this change? Or perhaps a link to the MOOSE issue and discussion that prompted it?

Reviewer 2: @lindsayad Could you remind what is it?

Reviewer 2: it interfered with attachment of geometric relationship managers

Reviewer 2: ```suggestion
[GridPartitioner.md] creates a uniform processor grid that overlays the mesh to be partitioned 
and then assigns all elements of the to-be-partitioned mesh within each cell of the grid
to the same processor. Previously, a native MOOSE mesh was explicitly created to represent
the uniform processor grid, and this could cause issues for a specific scenario where the native 
moose mesh interfered with attachment of geometric relationship managers.  This 
issue has been addressed by creating a virtual grid instead of a native moose mesh, and all 
operations for mesh element assignments are done using the virtual temporary grid only.
```

Reviewer 2: @cticenhour This makes sense?

Reviewer: I made a couple edits to the suggested change comment, and I think that's good to go. Thank you!

Reviewer 2: Right, I can not commit my suggestion. We need someone to click button. Or you could create a fresh suggestion, and then I will be able to  click button.

Reviewer: I'll create a fresh one, just a second.

## Default PETSc Configuration Change

> As more and more users explore MOOSE's massively parallel HPC features for large-size problems,
> we changed PETSc to have 64-bit integers. This change should not affect simulations' performance,
> and the motivation is to have MOOSE run large-size simulations with hundreds of millions of DoFs
> or billions of DoFs smoothly. Due to this change, libMesh's DoF index will be a 64-bit integer as well. 

Reviewer: ```suggestion
As more and more users explore MOOSE's massively parallel HPC features for large-size problems,
it was determined that PETSc should have 64-bit integers enabled by default. This change should not affect 
simulation performance, and the motivation is to allow MOOSE to smoothly run large-size simulations with 
hundreds of millions (or billions) of DoFs. Due to this change, libMesh's DoF index has been set to be a 64-bit integer as well. 
```
